fdbclient/MonitorLeader.actor.cpp

```diff
@@ -347,16 +415,26 @@ void ClusterConnectionString::parseKey( std::string const& key ) {
 std::string ClusterConnectionString::toString() const {
 	std::string s = key.toString();
 	s += '@';
-	for(int i=0; i<coord.size(); i++) {
-		if (i) s += ',';
-		s += coord[i].toString();
+	int i = 0;
+	for (; i < coord.size(); i++) {
+		if (_resolveResults.find(coord[i]) == _resolveResults.end()) {
+			if (i) {
+				s += ',';
+			}
+			s += coord[i].toString();
+		}
+	}
+	for (auto const& host : hosts) {
+		if (i) {
```

Comment:
This is a bug. If all coord are from hostnames, then there will be a comma right after the @. The fix should be
```
if (s.find('@') != s.length() - 1) {
    s += ',';
}